Packaging opam for FreeBSD shows strange error

Dear all,

I want to update the FreeBSD port of devel/ocaml-opam from 2.0.8 to 2.1.1. I use the [full tarball](https://github.com/ocaml/opam/releases/download/2.1.1/opam-full-2.1.1.tar.gz) -- i.e. where all dependencies are embedded as tarballs (since the build phase does not have access to the network). While doing that, I noticed that now a full OCaml compiler is bootstrapped. This is done via shell/bootstrap-ocaml.sh, which right at the top has needlessly an `exit 1` if neither "curl" nor "wget" is found. As the full tarball contains all tarballs, this shell script will never execute "curl"/"wget", and the shell script should not exit with 1.

For the FreeBSD package, I added a local patch https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=259667 (https://bugs.freebsd.org/bugzilla/attachment.cgi?id=229304&action=diff) -- but it would be nice to fix the bootstrap story of the full tarball before a next release. Since I'm not sure how the bootstrap story is put together, I'm not sure what an appropriate fix is (in FreeBSD, we as well pass `--disable-checks` to configure since otherwise (at least in earlier releases) opam refused to build without curl/wget).


Thanks for reading and have a nice weekend,

Hannes

I'm not sure I understand what you mean. An experiment I conduct without any `ocamlc` in my PATH:
```
opam-full-2.1.1$ ./configure 
checking for ocamlc... no
checking for ocaml... no
checking for ocamldep... no
checking for ocamlmktop... no
checking for ocamlmklib... no
checking for ocamldoc... no
checking for ocamlbuild... no
configure: error: You must install the OCaml compiler, at least version 4.02.3 (or run: make compiler)
```

Now, the packaged compiler fro FreeBSD is 4.05.0, if this is installed I get:
```
opam-full-2.1.1$ ./configure
checking for ocamlc... ocamlc
OCaml version is 4.05.0
OCaml library path is /usr/local/lib/ocaml
checking for ocamlopt... ocamlopt
checking for ocamlc.opt... ocamlc.opt
checking for ocamlopt.opt... ocamlopt.opt
checking for ocaml... ocaml
checking for ocamldep... ocamldep
checking for ocamldep.opt... ocamldep.opt
checking for ocamlmktop... ocamlmktop
checking for ocamlmklib... ocamlmklib
checking for ocamldoc... ocamldoc
checking for ocamldoc.opt... ocamldoc.opt
checking for ocamlbuild... no
checking OCaml Sys.os_type... Unix
```

Maybe I misunderstand how building opam-full should be done -- should there be a `make bootstrap ; ./configure ; make` done? If this is the case, it is pretty unconvential - would it be possible to achieve this with the common `./configure ; make`?

I also tried to run `./configure` without `--disable-checks` that then (if neither wget nor curl is found) calls our to `${MAKE:-make} -q -C src_ext has-archives`, resulting in:
```
gmake: *** No rule to make target 'archives/findlib-1.9.1.tar.gz', needed by 'has-archives'.  Stop.
```

That makes me wonder why findlib/ocamlbuild/topkg are in Makefile.sources as it seems they are not used for compilation?

--- a/.github/workflows/ci.ml
+++ b/.github/workflows/ci.ml
@@ -17,7 +17,7 @@ let ocamls = [
   (* Legacy versions - core libraries only *)
   "4.03.0"; "4.04.2"; "4.05.0"; "4.06.1"; "4.07.1"; 
   (* Fully supported versions *)
-   "4.08.1"; "4.09.1"; "4.10.2"; "4.11.2"; "4.12.1"; "4.13.1"
+   "4.08.1"; "4.09.1"; "4.10.2"; "4.11.2"; "4.12.1"; "4.13.1"; "4.14.0";
 ]
 
 (* Entry point for the workflow. Workflows are specified as continuations where

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -148,7 +148,7 @@ jobs:
     needs: Analyse
     strategy:
       matrix:
-        ocamlv: [ 4.08.1, 4.09.1, 4.10.2, 4.11.2, 4.12.1, 4.13.1 ]
+        ocamlv: [ 4.08.1, 4.09.1, 4.10.2, 4.11.2, 4.12.1, 4.13.1, 4.14.0 ]
       fail-fast: true
     steps:
     - name: Install bubblewrap
@@ -185,9 +185,9 @@ jobs:
       matrix:
         host: [ x86_64-pc-cygwin, i686-w64-mingw32, x86_64-w64-mingw32, i686-pc-windows, x86_64-pc-windows ]
         build: [ x86_64-pc-cygwin ]
-        ocamlv: [ 4.13.1 ]
+        ocamlv: [ 4.14.0 ]
         include:
-          - ocamlv: 4.13.1
+          - ocamlv: 4.14.0
             host: i686-pc-cygwin
             build: i686-pc-cygwin
       fail-fast: false
@@ -268,7 +268,7 @@ jobs:
     needs: Analyse
     strategy:
       matrix:
-        ocamlv: [ 4.13.1 ]
+        ocamlv: [ 4.14.0 ]
       fail-fast: true
     steps:
     - name: Checkout tree
@@ -341,7 +341,7 @@ jobs:
     needs: [ Analyse, Build-Linux ]
     strategy:
       matrix:
-        ocamlv: [ 4.13.1 ]
+        ocamlv: [ 4.14.0 ]
       fail-fast: false
     env:
       OPAM_TEST: 1
@@ -395,7 +395,7 @@ jobs:
     needs: Analyse
     strategy:
       matrix:
-        ocamlv: [ 4.13.1 ]
+        ocamlv: [ 4.14.0 ]
       fail-fast: false
     env:
       OPAM_TEST: 1
@@ -481,7 +481,7 @@ jobs:
     strategy:
       matrix:
         solver: [ z3, 0install ]
-        ocamlv: [ 4.13.1 ]
+        ocamlv: [ 4.14.0 ]
       fail-fast: false
     env:
       SOLVER: ${{ matrix.solver }}
@@ -529,7 +529,7 @@ jobs:
     strategy:
       matrix:
         solver: [ z3, 0install ]
-        ocamlv: [ 4.13.1 ]
+        ocamlv: [ 4.14.0 ]
       fail-fast: false
     env:
       SOLVER: ${{ matrix.solver }}
@@ -577,7 +577,7 @@ jobs:
     needs: [ Analyse, Build-Linux ]
     strategy:
       matrix:
-        ocamlv: [ 4.13.1 ]
+        ocamlv: [ 4.14.0 ]
       fail-fast: false
     steps:
     - name: Install bubblewrap
@@ -610,7 +610,7 @@ jobs:
     needs: [ Analyse, Build-macOS ]
     strategy:
       matrix:
-        ocamlv: [ 4.13.1 ]
+        ocamlv: [ 4.14.0 ]
       fail-fast: false
     steps:
     - name: Checkout tree

--- a/master_changes.md
+++ b/master_changes.md
@@ -223,6 +223,7 @@ users)
   * Cleared explanation of dependency vendoring in configure [#5277 @dra27 - fix #5271]
   * Switch autoconf required version to 2.71 [#5161 @dra27]
   * Remove src/client/no-git-version when calling make clean [#5290 @kit-ty-kate]
+  * Update the bootstrap compiler to 4.14.0 [#5250 @kit-ty-kate]
 
 ## Infrastructure
   * Fix caching of Cygwin compiler on AppVeyor [#4988 @dra27]

--- a/release/Makefile
+++ b/release/Makefile
@@ -3,7 +3,7 @@ VERSION = $(shell git describe $(TAG))
 OPAM_VERSION = $(subst -,~,$(VERSION))
 GIT_URL = ..
 
-OCAMLV = 4.13.1
+OCAMLV = 4.14.0
 # currently hardcoded in Dockerfile.in
 OCAML_URL = https://caml.inria.fr/pub/distrib/ocaml-$(basename $(OCAMLV))/ocaml-$(OCAMLV).tar.gz
 

--- a/src_ext/Makefile
+++ b/src_ext/Makefile
@@ -11,8 +11,8 @@ endif
 
 PATCH ?= patch
 
-URL_ocaml = https://caml.inria.fr/pub/distrib/ocaml-4.13/ocaml-4.13.1.tar.gz
-MD5_ocaml = a55ca12a4e6edf83cb4777abdb7b2f4d
+URL_ocaml = https://caml.inria.fr/pub/distrib/ocaml-4.14/ocaml-4.14.0.tar.gz
+MD5_ocaml = 57a4fa799114b3addbf31e42c4f4a7b3
 
 URL_flexdll = https://github.com/alainfrisch/flexdll/archive/0.40.tar.gz
 MD5_flexdll = e68f7311179fa7e09408825b362c5c5a

